src/include/rados/objclass.h

```diff
@@ -0,0 +1,183 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_OBJCLASS_OBJCLASS_PUBLIC_H
+#define CEPH_OBJCLASS_OBJCLASS_PUBLIC_H
+
+#ifdef __cplusplus
+
+#include <map>
+#include <set>
+#include "../include/buffer.h"
```

Comment:
one small thing: change this to ```#include "buffer.h"``` so it references the symlink in include/rados/buffer.h from the source tree, and the version installed by librados-dev when installed from packages (you've already got the dependency on that package so buffer.h will be installed)